lib/rules/template-curly-spacing.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * @fileoverview Rule to enforce spacing around embedded expressions of template strings
+ * @author Toru Nagashima
+ * @copyright 2016 Toru Nagashima. All rights reserved.
+ * See LICENSE file in root directory for full license.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+var astUtils = require("../ast-utils");
+
+//------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+var OPEN_PAREN = /\$\{$/;
+var CLOSE_PAREN = /^\}/;
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+    var sourceCode = context.getSourceCode();
+    var always = context.options[0] === "always";
+
+    /**
+     * Checks that a given TemplateElement node has spaces around itself.
+     * If no, this reports it.
+     *
+     * @param {ASTNode} node - A node to check. This is a TemplateElement.
+     * @returns {void}
+     */
+    function checkForAlways(node) {
+        var token = sourceCode.getFirstToken(node);
+        var prevToken, nextToken;
+
+        // Check spacing before `}`.
+        if (CLOSE_PAREN.test(token.value) &&
+            (prevToken = sourceCode.getTokenBefore(node)) &&
+            astUtils.isTokenOnSameLine(prevToken, token) &&
+            !sourceCode.isSpaceBetweenTokens(prevToken, token)
+        ) {
+            context.report({
+                node: node,
+                message: "Expected space(s) before '}'.",
+                fix: function(fixer) {
+                    return fixer.insertTextBefore(node, " ");
+                }
+            });
+        }
+
+        // Check spacing after `${`.
+        if (OPEN_PAREN.test(token.value) &&
+            (nextToken = sourceCode.getTokenAfter(node)) &&
+            astUtils.isTokenOnSameLine(token, nextToken) &&
+            !sourceCode.isSpaceBetweenTokens(token, nextToken)
+        ) {
+            context.report({
+                node: node,
+                loc: {
+                    line: node.loc.end.line,
+                    column: node.loc.end.column - 2
+                },
+                message: "Expected space(s) after '${'.",
+                fix: function(fixer) {
+                    return fixer.insertTextAfter(node, " ");
+                }
+            });
+        }
+    }
+
+    /**
+     * Checks that a given TemplateElement node has no space around itself.
+     * If yes, this reports it.
+     *
+     * @param {ASTNode} node - A node to check. This is a TemplateElement.
+     * @returns {void}
+     */
+    function checkForNever(node) {
```

Comment:
This function has an identical structure to `checkForAlways`. Can't they be combined?